### PR TITLE
adds documentation for credit_flow usage

### PR DESCRIPTION
**Pull request by videlalvaro**

### CODE REVIEW

File: src/credit_flow.erl [hunk 1]
```diff
@@ -27,6 +27,24 @@
 %% receiver it will not grant any more credit to its senders when it
 %% is itself blocked - thus the only processes that need to check
 %% blocked/0 are ones that read from network sockets.
+%%
+%% Credit flows left to right when process send messags down the
+%% chain, starting at the rabbit_reader, ending at the msg_store:
+%%  reader -> channel -> queue_process -> msg_store.
+%%
+%% If the message store has a back log, then it will block the
+%% queue_process, which will bloc the channel, and finally the reader
```
Comment (dumbbell) [inline src/credit_flow.erl]: `s/bloc/block/`

File: src/rabbit_amqqueue_process.erl [hunk 2]
```diff
@@ -662,9 +662,21 @@ handle_ch_down(DownPid, State = #q{consumers          = Consumers,
...
-                                            pmon:demonitor(DownPid, Senders)
+                                   false ->
+                                       Senders;
+                                   true  ->
+    %% A rabbit_channel process died. Here credit_flow will take care
+    %% of cleaning up the rabbit_amqqueue_process process dictionary
+    %% with regards to the credit we were tracking for the channel
+    %% process. See handle_cast({deliver, Deliver}, State) in this
+    %% module. In that cast function we process deliveries from the
+    %% channel, which means whe credit_flow:ack/1 said messages, so we
```
Comment (dumbbell) [inline src/rabbit_amqqueue_process.erl]: `s/whe/when/`? I don't understand this sentence.

Comment (videlalvaro) [inline src/rabbit_amqqueue_process.erl]: it's "we"

File: src/rabbit_channel.erl [hunk 3]
```diff
@@ -452,6 +463,12 @@ handle_info({'DOWN', _MRef, process, QPid, Reason}, State) ->
     State1 = handle_publishing_queue_down(QPid, Reason, State),
     State3 = handle_consuming_queue_down(QPid, State1),
     State4 = handle_delivering_queue_down(QPid, State3),
+    %% A rabbit_amqqueue_process has died. If our channel was being
+    %% blocked by this process, and no other process is blocking our
+    %% channel, then this channel will will be unblocked. This means
```
Comment (dumbbell) [inline src/rabbit_channel.erl]: `s/will will/will/`

**Comment by dumbbell** [comment]
Fixes #330.

**Pull request merged**